substrate/frame/support/procedural/src/construct_runtime/expand/composite_helper.rs

```diff
@@ -68,3 +68,34 @@ pub(crate) fn expand_variant(
 		}
 	}
 }
+
+pub(crate) fn expand_variant_count(
+	composite_name: &str,
+	path: &PalletPath,
+	instance: Option<&Ident>,
+) -> TokenStream {
+	let composite_name = quote::format_ident!("{}", composite_name);
+
+	if let Some(inst) = instance {
+		quote! {
+			#path::#composite_name::<#path::#inst>::VARIANT_COUNT
+		}
+	} else {
+		// Wrapped `<`..`>` means: use default type parameter for enum.
+		//
+		// This is used for pallets without instance support or pallets with instance support when
+		// we don't specify instance:
+		//
+		// ```nocompile
+		// pub struct Pallet<T, I = ()>{..}
+		//
+		// #[pallet::composite_enum]
+		// pub enum HoldReason<I: 'static = ()> {..}
+		//
+		// Pallet1: pallet_x,  // <- default type parameter
+		// ```
```

Comment:
Is nocompile needed here?

Reply:
@liamaharon The auto-merge happened faster than I could get to my PC. Thank you, I've created another PR to address your comments: https://github.com/paritytech/polkadot-sdk/pull/3150